src/app/components/ContactEmailSettingsModal.js

```diff
@@ -55,72 +57,62 @@ const ContactEmailSettingsModal = ({ userKeysList, contactID, properties, emailP
+        const revokedFingerprints = new Set();
+        const noPinnedKeyCanSend = pinnedKeys.reduce(async (acc, publicKey, index) => {
+            const fingerprint = publicKey.getFingerprint();
+            const { isExpired, isRevoked } = await getKeyEncryptStatus(publicKey);
+            trustedFingerprints.add(fingerprint);
+            isExpired && expiredFingerprints.add(fingerprint);
+            isRevoked && revokedFingerprints.add(fingerprint);
+            const cannotSend = isExpired || isRevoked;
+            return index ? cannotSend : acc && cannotSend;
+        }, undefined);
 
-                    if (field === 'x-pm-sign' && value) {
-                        acc.sign = value === 'true';
-                        return acc;
-                    }
-
-                    if (field === 'x-pm-scheme' && value) {
-                        acc.scheme = value;
-                        return acc;
-                    }
-
-                    if (field === 'x-pm-mimetype' && value) {
-                        acc.mimeType = value;
-                        return acc;
-                    }
-
-                    return acc;
-                },
-                { contactKeyPromises: [], mimeType: '', encrypt: false, scheme: '', sign: Sign === PGP_SIGN } // Default values
-            );
-        const contactKeys = (await Promise.all(contactKeyPromises)).filter(Boolean);
-        const config = await getPublicKeysEmailHelper(api, Email);
-        const internalUser = isInternalUser(config);
+        // prepare keys retrieved from the API
+        const apiKeysConfig = await getPublicKeysEmailHelper(api, Email);
+        const internalUser = isInternalUser(apiKeysConfig);
         const externalUser = !internalUser;
-        const apiKeys = config.publicKeys;
-        const [unarmoredKeys, keysExpired] = await Promise.all([
-            getRawInternalKeys(config),
-            allKeysExpired(contactKeys)
-        ]);
+        const verifyOnlyFingerprints = new Set();
+        const { apiKeys } = apiKeysConfig.Keys.reduce(
+            (acc, { Flags }, index) => {
+                const publicKey = apiKeysConfig.publicKeys[index];
+                if (publicKey) {
+                    acc.apiKeys.push(publicKey);
+                    const isVerificationOnly = !(Flags & KEY_FLAGS.ENABLE_ENCRYPTION);
+                    isVerificationOnly && verifyOnlyFingerprints.add(publicKey.getFingerprint());
+                }
+                return acc;
+            },
+            { apiKeys: [] }
+        );
+        const orderedApiKeys = sortApiKeys(apiKeys, trustedFingerprints, verifyOnlyFingerprints);
+        const trustedApiKeys = orderedApiKeys.filter((key) => trustedFingerprints.has(key.getFingerprint()));
+        const noTrustedApiKeyCanSend = !trustedApiKeys.length
+            ? false
+            : !trustedApiKeys.map((key) => !verifyOnlyFingerprints.has(key.getFingerprint())).filter(Boolean).length;
```

Comment:
```suggestion
            : !trustedApiKeys.some((key) => !verifyOnlyFingerprints.has(key.getFingerprint()))
```

Did you consider this?

Reply:
That's better indeed!